File: python/src/vo_ufadevconf/application/dto.py
from __future__ import annotations

from dataclasses import dataclass


@dataclass(frozen=True)
class CreatePickupPointCommand:
    code: str
    address: str
    min_weight_grams: int
    max_weight_grams: int


@dataclass(frozen=True)
class CheckParcelFitsQuery:
    pickup_point_code: str
    tracking_number: str
    contents_description: str
    parcel_weight_grams: int


@dataclass(frozen=True)
class ParcelFitResult:
    pickup_point_code: str
    tracking_number: str
    fits: bool

File: go/internal/application/dto.go
package application

// CreatePickupPointCommand contains data required to register a pickup point.
type CreatePickupPointCommand struct {
    Code           string
    Address        string
    MinWeightGrams int64
    MaxWeightGrams int64
}

// CheckParcelFitsQuery describes the scenario of checking a parcel.
type CheckParcelFitsQuery struct {
    PickupPointCode    string
    TrackingNumber     string
    ContentsDescription string
    ParcelWeightGrams  int64
}

// ParcelFitResult is returned by the application service.
type ParcelFitResult struct {
    PickupPointCode string
    TrackingNumber  string
    Fits            bool
}
